src/components/PeopleTable.tsx

```diff
@@ -1,683 +1,106 @@
-            >
-              Maria Sturm
-            </a>
-          </td>
-          <td>f</td>
-          <td>1835</td>
-          <td>1917</td>
-          <td>Seraphina Spelier</td>
-          <td>Charles Sturm</td>
-        </tr>
+type Props = {
+  people: Person[],
+};
 
-        <tr data-cy="person">
-          <td>
-            <a
-              className="has-text-danger"
-              href="#/people/emma-de-milliano-1876"
-            >
-              Emma de Milliano
-            </a>
-          </td>
-          <td>f</td>
-          <td>1876</td>
-          <td>1956</td>
-          <td>Sophia van Damme</td>
-          <td>Petrus de Milliano</td>
-        </tr>
+export const PeopleTable: React.FC<Props> = ({
+  people,
+}) => {
+  let visiblePeople = [...people];
+
+  const { searchParams } = useContext(SearchParamsContext);
+
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const currentSexFilter = searchParams.get('sex');
+  const currentQueryFilter = searchParams.get('query');
+  const currentCenturiesFilter = searchParams.getAll('centuries');
+
+  if (currentSexFilter) {
+    visiblePeople = visiblePeople.filter(({ sex }) => sex === currentSexFilter);
+  }
+
+  if (currentQueryFilter) {
+    visiblePeople = visiblePeople.filter(({ name }) => {
+      const nameLowercased = name.toLowerCase();
+      const currentQueryFilterLowercased = currentQueryFilter.toLowerCase();
+
+      return nameLowercased.includes(currentQueryFilterLowercased);
+    });
+  }
+
+  if (currentCenturiesFilter.length) {
+    visiblePeople = visiblePeople.filter(({ born, died }) => {
+      const bornCentury = Math.ceil(born / 1000);
+      const diedCentury = Math.ceil(died / 100);
+      const personLifeCenturies = [bornCentury, diedCentury];
+
+      return personLifeCenturies.some(century => (
+        currentCenturiesFilter.includes(String(century))
+      ));
+    });
+  }
+
+  if (sort) {
+    visiblePeople.sort((personA, personB) => {
+      switch (sort) {
+        case 'name':
+        case 'sex':
+          return personA[sort].localeCompare(personB[sort]);
+
+        case 'born':
+        case 'died':
+          return personA[sort] - personB[sort];
+
+        default:
+          return 0;
+      }
+    });
+
+    if (order) {
+      visiblePeople.reverse();
+    }
+  }
 
-        <tr data-cy="person">
-          <td>
-            <a href="#/people/emile-haverbeke-1877">Emile Haverbeke</a>
-          </td>
-          <td>m</td>
-          <td>1877</td>
-          <td>1968</td>
-          <td>
-            <a
-              className="has-text-danger"
-              href="#/people/maria-sturm-1835"
-            >
-              Maria Sturm
-            </a>
-          </td>
-          <td>
-            <a href="#/people/carolus-haverbeke-1832">Carolus Haverbeke</a>
-          </td>
-        </tr>
-      </tbody>
-    </table>
+  return (
+    visiblePeople.length
+      ? (
```

Comment:
Instead of checking for `visiblePeople.length`, it's better to use a conditional rendering with `visiblePeople.length > 0`. It makes the code more explicit and easier to understand. 🔍